src/main/scala/fi/oph/koski/valpas/repository/ValpasDatabaseService.scala

```diff
@@ -0,0 +1,106 @@
+package fi.oph.koski.valpas.repository
+
+import fi.oph.koski.config.KoskiApplication
+import fi.oph.koski.db.PostgresDriverWithJsonSupport.plainAPI._
+import fi.oph.koski.json.JsonSerializer
+import fi.oph.koski.log.Logging
+import fi.oph.koski.raportointikanta.ROpiskeluoikeusRow
+import fi.oph.koski.raportointikanta.RaportointiDatabaseSchema.{RHenkilöTable, ROpiskeluoikeusTable}
+import slick.jdbc.GetResult
+
+import java.sql.ResultSet
+
+class ValpasDatabaseService(application: KoskiApplication) extends Logging {
+  val db = application.raportointiDatabase
+  lazy val RHenkilöt = TableQuery[RHenkilöTable]
+  lazy val ROpiskeluoikeudet = TableQuery[ROpiskeluoikeusTable]
+
+  def getPeruskoulunValvojalleNäkyväOppija(oppijaOid: String, oppilaitosOids: Seq[String]) = {
+    // TODO: Kaipaa mahdollisesti optimointia, koska nyt haetaan kaikki oppijat (jotka oikeus nähdä) ja otetaan siitä se yksi oppija.
+    getPeruskoulunValvojalleNäkyvätOppijat(oppilaitosOids).find(_.henkilö.oid == oppijaOid)
+  }
+
+  def getPeruskoulunValvojalleNäkyvätOppijat(oppilaitosOids: Seq[String]): Seq[ValpasOppija] =
+    db.runDbSync(sql"""
+      SELECT
+        r_henkilo.oppija_oid,
+        r_henkilo.hetu,
+        r_henkilo.syntymaaika,
+        r_henkilo.etunimet,
+        r_henkilo.sukunimi,
+        json_agg(json_build_object(
+          'oid', r_opiskeluoikeus.opiskeluoikeus_oid,
+          'tyyppi', json_build_object(
+            'koodiarvo', r_opiskeluoikeus.koulutusmuoto,
+            'koodistoUri', 'opiskeluoikeudentyyppi'
+          ),
+          'oppilaitos', json_build_object(
+            'oid', r_opiskeluoikeus.oppilaitos_oid,
+            'nimi', json_build_object(
+              'fi', r_opiskeluoikeus.oppilaitos_nimi
+            )
+          ),
+          'alkamispäivä', r_opiskeluoikeus.alkamispaiva,
+          'päättymispäivä', r_opiskeluoikeus.paattymispaiva,
+          'ryhmä', r_opiskeluoikeus.luokka
+        )) opiskeluoikeudet
```

Comment:
Luokalle jääneellä 9-luokkalaisella voi varmaan olla samassa opiskeluoikeudessa monta 9-luokan päätason suoritusta, joka johtaa siihen, että tähän taulukkoon voi tulla monta riviä samalla opiskeluoikeus_oid:lla. Tässä tapauksessa myös niiden sisältö voi olla täsmälleen sama, jos oppija jatkaa samannimisessä ryhmässä luokallejääntinsä jälkeen. Kannattaa ainakin selvittää, miten luokallejäänti Kosken tietomallissa esitetään, ehkä ysiluokan suorituksia ei saakaan olla montaa ja se estetään validaatioilla tms.

Reply:
Luokallejääntitapaukselle joka tapauksessa kannattanee tehdä oma testi, se on kuitenkin tapaus, joka on aika helppo rikkoa tulevaisuudessakin, kun voi olla vähän yllättävä.

Reply:
Tässä pitäisi siis riittää, että lisätään tähän kyselytasolle päätason suorituksen oid mukaan. Ei mielestäni tarvitse ottaa vastuuta esitystavasta näin matalalla tasolla, eli näkymät saavat kaivaa itse opiskeluoikeuslistasta esim. viimeisimmän tai ryhmitellä ne haluamallaan tavalla.

Reply:
Joo, kuulostaa ihan hyvältä. Ehkä kentän voisi sitten nimetä myös joksikin muuksi kuin opiskeluoikeudet? Esim. opiskelutiedot, suoritustiedot tms.

Päätason suorituksen oidiin lisäksi mukaan voisi ehkä laittaa peruskoulun vuosiluokan suoritukselta myös mahdollisen vuosiluokan alkamispäivän ja mahdollisen suorituksen vahvistuspäivän, niin järjestäminen kälissä onnistuu, jos tarvitsee?